apps/deploy-web/src/pages/deploy-linux/index.tsx

```diff
@@ -1,5 +1,5 @@
-import { NewDeploymentContainer } from "@src/components/new-deployment/NewDeploymentContainer";
+import { SdlBuilderProvider } from "@src/context/SdlBuilderProvider/SdlBuilderProvider";
 
-export default function NewDeploymentPage() {
-  return <NewDeploymentContainer imageSource="ssh-vms" ssh />;
+export default function DeployLinuxPage() {
+  return <SdlBuilderProvider imageSource="ssh-vms" hiddenComponents={["command", "service-count", "ssh-toggle", "yml-editor", "yml-uploader"]} />;
```

Comment:
I'm not sure I like using context providers at the page level, because for instance we need to declare the provider on each page that would use one of the sub components of the SdlBuilder, which is currently 3 pages (new-deployment, deploy-linux, sdl-builder).

I think a better approach would be to use jotai and custom hook to set/access the global state.

Reply:
Yea, I know and I'm going to add this to other pages. This whole thing is already overly complicated. Context is an in built thing that imo fits this perfectly as it's not really needed globally but within the context of a certain page (every page would have own context). Unlike state managers are needed for global state.

Reply:
I think using the Context api is fine. I always use it at the layout level to avoid having to repeat the provider at multiple places, especially if the component is hardcoded. Perhaps we could just put `{children}` and let the page call it's own children.

Reply:
Agreed

Reply:
I didn't do that just coz haven't thought of other pages. After Anil's comment in slack I realised I'd need to do that

Reply:
Makes sense!